components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeConfiguration.java

```diff
@@ -46,19 +82,57 @@ public void setName(String name) {
         this.name = name;
     }
 
+    public String getService() {
+        return service;
+    }
+
+    public void setService(String service) {
+        Assert.isTrue(supportedServices.contains(service), "Unsupported service, we have: " + supportedServices);
```

Comment:
We tend to do this kind of validation in the doStart method of the component or when creating the producer / consumer (if specific to them) and there is a `ObjectHelper.notNull / notEmpty` we tend to use for this kind.

Reply:
Yes, I saw the ObjectHelper in Camel. The XChange library however also provides Assert.isTrue() which I couldn't find in Camel